myia/utils.py

```diff
@@ -93,6 +73,74 @@ def list_str(lst: List):
     return f'[{elements}]'
 
 
+class TypeMap(dict):
```

Comment:
That isn't true. All dict methods are fine to use on `TypeMap`, although there can be issues if you mutate it after use.

Reply:
Not true:

```python
from myia.utils import TypeMap

tm = TypeMap({object: 'info'})
v1 = tm.get(str)  # returns None
v2 = tm[str]      # returns 'info'
```

Maybe implementing `__missing__` instead of `__getitem__` would help, but I'm not sure it does what you want.